### PR TITLE
umbrella chart with ingress and storage example

### PR DESCRIPTION
**Pull request by survivant**
I hope that will help with the issue : https://github.com/openebs/monitoring/issues/92

### CODE REVIEW

File: docs/extends-stack-with-umbrella-chart.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
+# Extending the monitoring stack
+
+### Prerequisite
+
+- Helm 3+ installed
+- OpenEBS installed
+
+#What is a umbrella chart and why use it ?
```
Comment (Ab-hishek) [inline docs/extends-stack-with-umbrella-chart.md]: Add a space after `#`.

File: docs/extends-stack-with-umbrella-chart.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+┣ 📜Chart.yaml
+┗ 📜values.yaml
+```
+
+| File               | Description                                                                      |
+| ------             | ------                                                                           |
+| Chart.yaml         | Contains your chart identification with a dependency to OpenEBS monitoring stack |
+| values.yaml        | Contains all your settings values                                                |
+
+#Umbrella chart with ingress and storage enabled
```
Comment (Ab-hishek) [inline docs/extends-stack-with-umbrella-chart.md]: Add a space after `#`.

File: docs/extends-stack-with-umbrella-chart.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+### Prerequisite
+
+- Helm 3+ installed
+- OpenEBS installed
+
+#What is a umbrella chart and why use it ?
+
+An umbrella chart is a way of encapsulate multiple complex charts with one chart.
+
+>Many of the charts in the CNCF Artifact Hub are "building blocks" for creating more advanced applications. But charts may be used to create instances of large-scale applications. In such cases, a single umbrella chart may have multiple subcharts, each of which functions as a piece of the whole.
```
Comment (Ab-hishek) [inline docs/extends-stack-with-umbrella-chart.md:12]: Do we want the formatting to be like this or as a list?

File: docs/extends-stack-with-umbrella-chart.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+| Chart.yaml         | Contains your chart identification with a dependency to OpenEBS monitoring stack |
+| values.yaml        | Contains all your settings values                                                |
+
+#Umbrella chart with ingress and storage enabled
+
+Here a example that will enable ingress and storage for OpenEBS monitoring stack.
+
+In the file **Chart.yaml**, you will enter your umbrella chart information with a dependency to OpenEBS monitoring stack.  The property name : `condition` is optional.  That property will allow you to install that dependency or not depending of the value in `values.yaml`. 
+
+####Chart.yaml
```
Comment (Ab-hishek) [inline docs/extends-stack-with-umbrella-chart.md]: Add a space after `#`.

File: docs/extends-stack-with-umbrella-chart.md [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+    condition: openebs-monitoring.install
+````
+
+In the file **values.yaml**, you will define your configuration for the monitoring stack.  
+
+If you defined a **dependency condition** in **Chart.yaml**, you will have to define the value in that condition.  In this example, the condition name is `openebs-monitoring.install`.  
+
+if you provide only the condition value, you will deploy OpenEBS monitoring stack with the default values provided in there chart.
+
+####value.yaml (with default values)
```
Comment (Ab-hishek) [inline docs/extends-stack-with-umbrella-chart.md]: Add a space after `#`.

File: docs/extends-stack-with-umbrella-chart.md [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+````yaml
+openebs-monitoring:
+  install: true
+````
+
+####value.yaml (with ingress and storage)
```
Comment (Ab-hishek) [inline docs/extends-stack-with-umbrella-chart.md]: Add a space after `#`.

File: docs/extends-stack-with-umbrella-chart.md [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+        hosts:
+          - dev.dynamic.cluster109.local
+
+# NDP SECTION
+  ## Configuration for node-problem-detector subchart
+  node-problem-detector:
+    install: true
+````
+
+#How to deploy the umbrella chart
```
Comment (Ab-hishek) [inline docs/extends-stack-with-umbrella-chart.md]: Add a space after `#`.

File: docs/extends-stack-with-umbrella-chart.md [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+ ┃ ┗ 📜openebs-monitoring-0.4.8.tgz
+ ┣ 📂templates
+ ┣ 📜Chart.lock
+ ┣ 📜Chart.yaml
+ ┗ 📜values.yaml
+```
+
+### Install the chart
+
+####From the umbrella folder
```
Comment (Ab-hishek) [inline docs/extends-stack-with-umbrella-chart.md]: Add a space after `#`.

File: docs/extends-stack-with-umbrella-chart.md [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+```console
+helm install [RELEASE_NAME] . -n  [NAMESPACE] --create-namespace
+```
+
+ex :
+````console
+helm install monitoring . -n monitoring --create-namespace
+````
+
+####From a helm chart repository
```
Comment (Ab-hishek) [inline docs/extends-stack-with-umbrella-chart.md]: Add a space after `#`.

File: docs/extends-stack-with-umbrella-chart.md [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+````yaml
+openebs-monitoring:
+  install: true
+````
+
+####value.yaml (with ingress and storage)
+
+To use storage for the monitoring stack, you will have to create the storage class before using it.  In this example, I use a storage class named : `sc-metrics`.
+
+For ingress, I used a domain name : `dev.dynamic.cluster109.local`
```
Comment (Ab-hishek) [inline docs/extends-stack-with-umbrella-chart.md]: Can we change the language here from I to a more generalised context?

File: docs/extends-stack-with-umbrella-chart.md (hunk 3 above)
Comment (survivant) [reply docs/extends-stack-with-umbrella-chart.md:12]: it's a "quote"

File: docs/extends-stack-with-umbrella-chart.md (hunk 10 above)
Comment (survivant) [reply docs/extends-stack-with-umbrella-chart.md]: no trouble at all.  english is not my native language, so I accept anything better than what I wrote :)

File: docs/extends-stack-with-umbrella-chart.md [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+
+#### value.yaml (with default values)
+````yaml
+openebs-monitoring:
+  install: true
+````
+
+#### value.yaml (with ingress and storage)
+
+To use storage for the monitoring stack, you will have to create the storage class before using it.  In this example, I use a storage class named : `sc-metrics`.
```
Comment (kmova) [inline docs/extends-stack-with-umbrella-chart.md]:
```suggestion
To use storage for the monitoring stack, you will have to create the storage class before using it.  In the following example, I use a storage class named: `sc-metrics` which already exists in my cluster.
```

File: docs/extends-stack-with-umbrella-chart.md [hunk 12]
```diff
@@ -0,0 +1,235 @@
...
+
+For ingress, I used a domain name : `dev.dynamic.cluster109.local`
```
Comment (kmova) [inline docs/extends-stack-with-umbrella-chart.md]:
```suggestion
Similarly you can also customize other parameters. For instance, if domain name has to be used to access the user interfaces, the `ingress` section needs to be updated. In the following example I use the value: `dev.dynamic.cluster109.local`. 
```

**Review by kmova — APPROVED** [review]

**Pull request merged**